c10/util/Half.h

```diff
@@ -289,6 +314,12 @@ C10_HOST_DEVICE inline float fp16_ieee_to_fp32_value(uint16_t h) {
  * between integer and floating-point variables.
  */
 inline uint16_t fp16_ieee_from_fp32_value(float f) {
+#if defined(NATIVE_FP16)
+  _Float16 fp16 = static_cast<_Float16>(f);
+  return reinterpret_cast<uint16_t>(fp16);
+#elif defined(X86_F16)
+  return _cvtss_sh(f, _MM_FROUND_TO_NEAREST_INT);
```

Comment:
Where is `X86_F16` defined? I guess you need `at::vec::float2half_scalar(x)` here?